include/modem/at_cmd.h

```diff
@@ -132,6 +132,27 @@ int at_cmd_write(const char *const cmd,
 		 size_t buf_len,
 		 enum at_cmd_state *state);
 
+/**
+ * @brief Send a formatted AT command.
+ *
+ * @param fmt Command format.
+ * @param ... Format parameters.
+ * @return int Zero on success, a negative errno otherwise.
+ */
+int at_cmd_printf(const char *fmt, ...);
```

Comment:
Not having `enum at_cmd_state *state` as a parameter is somewhat dangerous as zero is a valid error code too for AT_CMD_ERROR_CME. People should always check whether error occurred with` if (state != AT_CMD_OK)`. I know this is not the case and actually quite rare in our code base.